OpenRA.Mods.Common/Widgets/Logic/Ingame/ProductionTooltipLogic.cs

```diff
@@ -22,13 +22,13 @@ namespace OpenRA.Mods.Common.Widgets.Logic
 	public class ProductionTooltipLogic : ChromeLogic
 	{
 		[ObjectCreator.UseCtor]
-		public ProductionTooltipLogic(Widget widget, TooltipContainerWidget tooltipContainer, ProductionPaletteWidget palette, World world)
+		public ProductionTooltipLogic(Widget widget, TooltipContainerWidget tooltipContainer, World world, Player player, Func<ProductionIcon> getTooltipIcon)
 		{
-			var mapRules = palette.World.Map.Rules;
-			var pm = palette.World.LocalPlayer.PlayerActor.Trait<PowerManager>();
-			var pr = palette.World.LocalPlayer.PlayerActor.Trait<PlayerResources>();
+			var mapRules = world.Map.Rules;
```

Comment:
You could use `player.World.Map.Rules` and then don't need the extra argument above.